release.py

```diff
@@ -0,0 +1,113 @@
+#! /usr/bin/env python
+# Tahoe-LAFS -- secure, distributed storage grid
+#
+# Copyright © 2006-2012 The Tahoe-LAFS Software Foundation
+#
+# This file is part of Tahoe-LAFS.
+#
+# See the docs/about.rst file for licensing information.
+import os
+import glob
+import shutil
+import sys
+import subprocess
+import argparse
+import re
+
+class bcolors:
+    HEADER = '\033[95m'
+    OKBLUE = '\033[94m'
+    OKCYAN = '\033[96m'
+    OKGREEN = '\033[92m'
+    WARNING = '\033[93m'
+    FAIL = '\033[91m'
+    ENDC = '\033[0m'
+    BOLD = '\033[1m'
+    UNDERLINE = '\033[4m'
+
+
+parser = argparse.ArgumentParser()
+parser.add_argument(
+    "--clean", action="store_true", help="Cancel existing release process, clean files"
+)
+parser.add_argument(
+    "--ignore-deps", action="store_true", help="Ignore dependency checks"
+)
+parser.add_argument("--fin", action="store_true", help="Finish release proccess")
+parser.add_argument("--sign", type=str, help="Signing key")
+parser.add_argument("--ticket", type=int, help="Ticket number", required=True)
+parser.add_argument("--tag", type=str, help="Release tag", required=True)
+args = parser.parse_args()
+
+# looks like XXXX.release-1.16.0
+BRANCH = "{ticket}.release-{tag}".format(ticket=args.ticket, tag=args.tag)
+RELEASE_TITLE = "Release {tag} ({date})".format(tag=args.tag, date="")
+RELEASE_FOLDER = "../tahoe-release-{0}".format(args.tag)
+CONTINUE_INSTRUCTION = bcolors.BOLD + "Instruction : run ./venv/bin/python release.py --ignore-deps --tag {tag} --ticket {ticket} --sign {sign} -fin".format(
+    tag=args.tag, ticket=args.ticket, sign=args.sign
+) + bcolors.ENDC
+
+
+def clean():
+    for release_dir in glob.glob("../tahoe-release-*"):
```

Comment:
This makes me nervous (globbing a thing that's going to get deleted completely).

Reply:
It one of the few options available for deleting wildcards a python script?

Reply:
e.g. what if I had a directory called `tahoe-release-DO-NOT-DELETE`?

Reply:
@meejah I got your point... Don't worry about that I am going to clean only things related the parameters passed.  Would address all reviews and clean this up so we can talk about it on monday.